src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
+
+  static addAliveAnimals(item) {
+    Animal.alive.push(item);
+  }
+
+  static removeAnimals() {
+    Animal.alive = Animal.alive.filter((item) => item.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+    Animal.addAliveAnimals(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.addAliveAnimals(this);
+  }
+
+  bite(item) {
```

Comment:
The `bite` method should check if the `item` is an instance of `Carnivore` and return early to prevent a carnivore from biting another carnivore, as per the task requirements.